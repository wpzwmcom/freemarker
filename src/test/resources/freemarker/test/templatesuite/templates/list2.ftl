<@testList ["aardvark", "bear", "cat", "dog"] />

<@testList ["aardvark"] />

<@testList [] />

<@testList listables.list />

<@testList listables.set />

<@testList listables.getIterator />

<@testList listables.emptyList />

<@testList listables.emptySet />

<@testList listables.getEmptyIterator />

<#macro testList xs>
=== [${resolve(xs)?join(", ")}] ===
<#assign resolveCallCnt = 0>

-- List+sep:
<#list resolve(xs) as x>
    ${x}<#sep>,</#sep>
</#list>
-- List+else:
<#list resolve(xs) as x>
    ${x}
<#else>
    Empty!
</#list>
-- List+items:
<#list resolve(xs)>
    [
    <#items as x>
        ${x!'U'}
    </#items>
    ]
</#list>
-- List+items+else:
<#list resolve(xs)>
    [
    <#items as x>
        ${x!'U'}
    </#items>
    ]
<#else>
    Empty!
</#list>
-- List+items+sep+else:
<#list resolve(xs)>
    [
    <#items as x>
        ${x!'U'}<#sep>,</#sep>
    </#items>
    ]
<#else>
    Empty!
</#list>
<@assertEquals expected=5 actual=resolveCallCnt />
--
</#macro>

<#function resolve xs>
    <#assign resolveCallCnt = (resolveCallCnt!0) + 1>
    <#if xs?isMethod>
        <#return xs()>
    <#else>
        <#return xs>
    </#if>
</#function>