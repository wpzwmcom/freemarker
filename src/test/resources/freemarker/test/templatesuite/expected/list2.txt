=== [aardvark, bear, cat, dog] ===

-- List+sep:
    aardvark,
    bear,
    cat,
    dog
-- List+else:
    aardvark
    bear
    cat
    dog
-- List+items:
    [
        aardvark
        bear
        cat
        dog
    ]
-- List+items+else:
    [
        aardvark
        bear
        cat
        dog
    ]
-- List+items+sep+else:
    [
        aardvark,
        bear,
        cat,
        dog
    ]
--

=== [aardvark] ===

-- List+sep:
    aardvark
-- List+else:
    aardvark
-- List+items:
    [
        aardvark
    ]
-- List+items+else:
    [
        aardvark
    ]
-- List+items+sep+else:
    [
        aardvark
    ]
--

=== [] ===

-- List+sep:
-- List+else:
    Empty!
-- List+items:
-- List+items+else:
    Empty!
-- List+items+sep+else:
    Empty!
--

=== [11, 22, 33] ===

-- List+sep:
    11,
    22,
    33
-- List+else:
    11
    22
    33
-- List+items:
    [
        11
        22
        33
    ]
-- List+items+else:
    [
        11
        22
        33
    ]
-- List+items+sep+else:
    [
        11,
        22,
        33
    ]
--

=== [11, 22, 33] ===

-- List+sep:
    11,
    22,
    33
-- List+else:
    11
    22
    33
-- List+items:
    [
        11
        22
        33
    ]
-- List+items+else:
    [
        11
        22
        33
    ]
-- List+items+sep+else:
    [
        11,
        22,
        33
    ]
--

=== [11, 22, 33] ===

-- List+sep:
    11,
    22,
    33
-- List+else:
    11
    22
    33
-- List+items:
    [
        11
        22
        33
    ]
-- List+items+else:
    [
        11
        22
        33
    ]
-- List+items+sep+else:
    [
        11,
        22,
        33
    ]
--

=== [] ===

-- List+sep:
-- List+else:
    Empty!
-- List+items:
-- List+items+else:
    Empty!
-- List+items+sep+else:
    Empty!
--

=== [] ===

-- List+sep:
-- List+else:
    Empty!
-- List+items:
-- List+items+else:
    Empty!
-- List+items+sep+else:
    Empty!
--

=== [] ===

-- List+sep:
-- List+else:
    Empty!
-- List+items:
-- List+items+else:
    Empty!
-- List+items+sep+else:
    Empty!
--

